exporter/awss3exporter/exporter.go

```diff
@@ -72,7 +72,16 @@ func (e *s3Exporter) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) err
 		return err
 	}
 
-	return e.uploader.Upload(ctx, buf)
+	s3Prefix := ""
+	if s3PrefixKey := e.config.OTelAttrsToS3.S3Prefix; s3PrefixKey != "" {
+		if value, ok := md.ResourceMetrics().At(0).Resource().Attributes().Get(s3PrefixKey); ok {
```

Comment:
I'd encourage you to manage this similar to splunk's version [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/splunkhecexporter/logdata_to_splunk.go#L88). 

Breaking it out into a different function, and querying resource attributes, then log attributes, would make sense. Alternatively, simply renaming this to `ResourceAttrstoS3` or similar to show clarity would help make it clear how this is mapped as is.